devel/release.md

```diff
@@ -1,8 +1,71 @@
 # Release process
 
-The release process of the python client involves creating (or updating) a release
-branch, updating release tags, and creating distribution packages and uploading them to
-pypi.
+The release process for the python client involves creating (or updating) a
+release branch, updating release tags, and creating distribution packages and
+uploading them to pypi.
+
+There are several releases per version:
+- snapshot
+- a1 (alpha release)
+- b1 (beta release)
+- final release
+
+Between each release, there is a waiting period of about two weeks for users to
+report issues. Typically, there is a single alpha or beta release, but if there
+are a higher than expected number of issues there can be multiple releases
+(e.g, a2 or b2).
+
+## Create or update release branch
+
+The release branch name should have release-x.x format. All minor and pre-releases
+should be on the same branch. To update an existing branch with master (only for
+latest pre-release):
+
+```bash
+export RELEASE_BRANCH=release-x.y
+git checkout $RELEASE_BRANCH
+git fetch upstream
+git rebase upstream/$RELEASE_BRANCH
+git pull upstream master
+```
+
+You may need to fix some conflicts. For auto-generated files, you can commit
+either version. They will be updated to the current version in the next step.
+
+## Update release tags
+
+Release tags are in the "scripts/constants.py" file. These are the constants you
+may need to update:
+
+CLIENT_VERSION: Client version should follow x.y.zDn where x,y,z are version
+numbers (integers) and D is one of "a" for alpha or "b" for beta and n is the
+pre-release number. For a final release, the "Dn" part should be omitted.
+Examples:
+- 1.0.0a1 (alpha release)
+- 2.0.1b2 (beta release)
+- 1.5.1 (final release)
+
+DEVELOPMENT_STATUS: Update it to one of the values of "Development Status" in
+[this list](https://pypi.python.org/pypi?%3Aaction=list_classifiers).
+
+After changing constants to proper versions, update the client using this
+command:
+
+```bash
+scripts/update-client.sh
+```
+
+Remove the tests directory (ref: https://github.com/kubernetes-client/python/commit/ec9c944f076999543cd2122aff2d86f969d82548).
```

Comment:
Could you move this line to the "hot issues" section, and link to the upstream issue https://github.com/OpenAPITools/openapi-generator/issues/5377?